### PR TITLE
[Core]Using base methods for determinant and inverse of the geometry jacobian

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by jcotela** [comment]
@loumalouomega I think that quadrilateral_3d_9 has the same problem.

**Comment by loumalouomega** [comment]
@jcotela , Ok

**Comment by loumalouomega** [comment]
Done @jcotela , can you check this @josep-m-carbonell ?

**Comment by roigcarlo** [comment]
@loumalouomega If this works would yo mind making a separate Pr for the Release Branch?

**Comment by loumalouomega** [comment]
Can no just be merged?

**Comment by roigcarlo** [comment]
we have to make a separate one for the release. It dosn't matter I will make a separate PR with some other fixes :+1:

**Comment by josep-m-carbonell** [comment]
It does not compile in my computer.

**Comment by roigcarlo** [comment]
Mine neither...

**Comment by roigcarlo** [comment]
It should work now

**Comment by josep-m-carbonell** [comment]
Ok now it works

**Review by jcotela — APPROVED** [review]

**Comment by loumalouomega** [comment]
Can we close this then?

**Comment by roigcarlo** [comment]
No. I have to make another Pr for the release...

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -2088,7 +2088,7 @@ class Geometry : public PointerVector<TPointType>
             if(rResult[pnt].size1() != this->WorkingSpaceDimension() ||  rResult[pnt].size2() != this->LocalSpaceDimension())
                 rResult[pnt].resize( (*this).size(), this->LocalSpaceDimension(), false );
             this->Jacobian(J,pnt, ThisMethod);
-            MathUtils<double>::InvertMatrix( J, Jinv, DetJ );
+            MathUtils<double>::GeneralizedInvertMatrix( J, Jinv, DetJ );
```
Comment (RiccardoRossi) [inline kratos/geometries/geometry.h:2091]: guys this is more expensive... i would say that 2D geometries in 3D space should reimplement it...

**Comment by loumalouomega** [comment]
The generalized inverse checks the size before compute, so if the system is
square performs the standard inverse

**Comment by RiccardoRossi** [comment]
@loumalouomega the point is that by doing this the behaviour was changed (an error was thrown before, and the error did make sense...). 

Aside on the behavioural change (which i think we should have discussed), even if we go for the generalized inverse i would use only there where it is needed, that is, I can understand that say Triangle3D3N uses the generalized inverse, however a generic geometry does not need to. So instead of changing the base class and removing the overrides, it may be better to reimplement in the overrided classes.

**Comment by loumalouomega** [comment]
But when the matrix is square it uses the regular inverse, it only adds an
extra check, nothing more

**Pull request merged**